code/obj/item/tool/Electronics.dm

```diff
@@ -917,6 +925,40 @@
 			boutput(user, "<span class='alert'>You need to use some tools on [target] before it can be deconstructed.</span>")
 			return
 
+ // here be surgeries
+	attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
+
+		if(!saw_surgery(M,user)) // if it ain't surgery I don't care
+			return ..()
+
+		if(prob(20))// doing surgery with a buzzsaw isn't a good idea
+			user.visible_message("<span class='alert'><b>[user]</b> messes up and injures [himself_or_herself(user)] with the [src]! </span>")
```

Comment:
So an important part of surgery is looking at the chat to see whether you failed a step, and I'm pretty sure if implemented like this you could succeed in a step and then get the message "messes up and injures themselves" which is going to make you think you failed the step and try again, thus hitting your patient with a buzzsaw unnecessarily.
I think a way to prevent this would be to call `surgeryCheck` to see if you're going to attempt surgery, then roll the extra buzzsaw fail chance and not call `saw_surgery` if you fail. That way you can't succeed and then be told you messed up.